File: VideoGameCatalogue.API.Tests/TestData.cs
using VideoGameCatalogue.API.Contracts;
using VideoGameCatalogue.API.Models;

namespace VideoGameCatalogue.API.Tests
{
    /// <summary>
    /// Builders for valid test objects. Tests override only the fields they care about,
    /// e.g. <c>TestData.ValidRequest() with { Title = "" }</c>.
    /// </summary>
    public static class TestData
    {
        public static VideoGameRequest ValidRequest() => new()
        {
            Title = "Elden Ring",
            Developer = "FromSoftware",
            Publisher = "Bandai Namco",
            ReleaseDate = new DateOnly(2022, 2, 25),
            Genre = "RPG",
            Price = 59.99m,
            Platform = "PC",
            MetacriticScore = 96,
            Description = "An action role-playing game."
        };

        public static VideoGame ValidGame(string title = "Test Game") => new()
        {
            Title = title,
            Developer = "Test Developer",
            Publisher = "Test Publisher",
            ReleaseDate = new DateOnly(2023, 1, 1),
            Genre = "Action",
            Price = 29.99m,
            Platform = "PC",
            MetacriticScore = 80,
            Description = "A test game."
        };
    }
}

File: VideoGameCatalogue.Frontend/src/app/testing/test-data.ts
import { Lookups, VideoGame, VideoGameRequest } from '../models/video-game.model';

export function aVideoGameRequest(overrides: Partial<VideoGameRequest> = {}): VideoGameRequest {
  return {
    title: 'The Legend of Zelda: Tears of the Kingdom',
    developer: 'Nintendo EPD',
    publisher: 'Nintendo',
    releaseDate: '2023-05-12',
    genre: 'Action-Adventure',
    price: 69.99,
    platform: 'Nintendo Switch',
    metacriticScore: 98,
    description: 'An epic adventure.',
    ...overrides
  };
}

export function aVideoGame(overrides: Partial<VideoGame> = {}): VideoGame {
  return {
    id: 1,
    ...aVideoGameRequest(),
    createdAt: '2024-01-01T00:00:00Z',
    updatedAt: '2024-01-01T00:00:00Z',
    ...overrides
  };
}

export const testLookups: Lookups = {
  genres: ['Action-Adventure', 'RPG'],
  platforms: ['Nintendo Switch', 'PC']
};
